### PR TITLE
Add NAMES

### PR DESCRIPTION
**Pull request by piecubed**

### CODE REVIEW

File: miniirc_discord.py [hunk 1]
```diff
@@ -86,7 +86,18 @@ def quote(self, *msg, force = None):
...
             self._run(self._client.change_presence(game = game))
-
+        elif cmd in ('EMBED_NOTICE',):
+            if len(args) == 4:
+                EMBED_NOTICE = discord.Embed(title=args[0], description=args[1], colour=arg[2])
+                EMBED_NOTICE.set_author(name=args[3], icon_url=discord.client.user.default_avatar_url)
+            else:
+                self.debug('Invalid call to EMBED_NOTICE')
+        elif cmd in ('NAMES',):
+            discord.Member._dict_
```
Comment (luk3yx) [inline miniirc_discord.py]: This does nothing and will probably throw an error.

File: miniirc_discord.py [hunk 2]
```diff
@@ -86,7 +86,18 @@ def quote(self, *msg, force = None):
             self.debug('Changing online presence:', game)
             game = discord.Game(name = game)
             self._run(self._client.change_presence(game = game))
-
+        elif cmd in ('EMBED_NOTICE',):
```
Comment (luk3yx) [inline miniirc_discord.py]: This should follow the [IRC](https://tools.ietf.org/html/rfc1459) [RFCs](https://tools.ietf.org/html/rfc2812) as closely as possible.

File: miniirc_discord.py [hunk 3]
```diff
@@ -86,7 +86,18 @@ def quote(self, *msg, force = None):
...
+            if len(args) == 4:
+                EMBED_NOTICE = discord.Embed(title=args[0], description=args[1], colour=arg[2])
```
Comment (luk3yx) [inline miniirc_discord.py]: `arg`?

**Review by luk3yx — CHANGES_REQUESTED** [review]

File: miniirc_discord.py (hunk 3 above)
Comment (luk3yx) [reply miniirc_discord.py]: This function also apparently does nothing.

Comment (luk3yx) [reply miniirc_discord.py]: This function also appears to do nothing.

File: miniirc_discord.py (hunk 2 above)
Comment (piecubed) [reply miniirc_discord.py]: ### GAH

File: miniirc_discord.py (hunk 1 above)
Comment (piecubed) [reply miniirc_discord.py]: ### FIXED

File: miniirc_discord.py [hunk 4]
```diff
@@ -90,14 +90,17 @@ def quote(self, *msg, force = None):
             if len(args) == 4:
                 EMBED_NOTICE = discord.Embed(title=args[0], description=args[1], colour=arg[2])
                 EMBED_NOTICE.set_author(name=args[3], icon_url=discord.client.user.default_avatar_url)
+                self._run(self._client.send_messge(chan, EMBED_NOTICE))
             else:
                 self.debug('Invalid call to EMBED_NOTICE')
         elif cmd in ('NAMES',):
-            discord.Member._dict_
-    
+            self._run(self._client.send_messge(chan, discord.Server.members))
```
Comment (luk3yx) [inline miniirc_discord.py]: `NAMES` normally replies with `RPL_NAMREPLY` and `RPL_ENDOFNAMES` (see [the RFC](https://tools.ietf.org/html/rfc1459#section-4.2.5)).

**Review by luk3yx — CHANGES_REQUESTED** [review]

**Pull request closed without merging**